generic.go

```diff
@@ -423,3 +423,19 @@ func GroupBy[V any, T comparable](slice []V, f func(v V) T) map[T][]V {
 
 	return result
 }
+
+// Partition returns two slices: the first containing all elements that returned
+// `true` for `f`, and the second containing all others
+func Partition[V any](slice []V, f func(v V) bool) ([]V, []V) {
+	pass, reject := []V{}, []V{}
```

Comment:
Do y'all have a better name for `pass` and `reject`?

Reply:
maybe `rest`?